esmvaltool/_recipe.py

```diff
@@ -210,6 +208,16 @@ def _update_target_grid(variable, variables, settings, config_user):
         parse_cell_spec(settings['regrid']['target_grid'])
 
 
+def _update_regrid_time(variable, settings):
+    """Input data frequency automatically for regrid_time preprocessor."""
+    regrid_time = settings.get('regrid_time')
+    if regrid_time is None:
+        return
```

Comment:
```suggestion
```
this is not needed, I think

Reply:
it is, because I want to keep the user's choice of passing in `frequency` in recipe, without it if there is no `regrid_time` in recipe, the code returns a `KeyError`

Reply:
Try it first

Reply:
I did! It's also a bitty bit obvious no? :grin:

Reply:
I was sleeping ;-)

Reply:
that's okay, B-man, catch some sleep you missed from home :grin: